services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -845,6 +869,29 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
         }
     }
 
+    /**
+     * Create action token 
+     * 
+     * A token to perform a set of required actions. The clientId parameters are
+     * optional.
+     * 
+     * @param redirectUri Redirect uri
+     * @param clientId Client id
+     * @param lifespan Number of seconds after which the generated token expires
+     * @param actions required actions the user needs to complete
+     * @return
+     */
+    @Path("generate-action-token")
+    @PUT
```

Comment:
Why PUT rather than POST for the creation of the action token ? If each call generates a new token, POST may be more appropriate.

Reply:
I expected that the function replaced the action token. Then it would be in my opinion PUT the method.
But we can rename them if POST is the better method.

Reply:
Good point from @fidgi. Could you change this?

Reply:
Changed